File: CPP/Pengiriman.cpp
#include <bits/stdc++.h>
#include "Kurir.cpp"
#pragma once

// Forward declare Kurir class
class Kurir;

using namespace std;

class Pengiriman {
private:
    string IdPengiriman;
    string StatusPengiriman;
    vector<Kurir> kurirList;
    float BeratPaket;
    string JenisPaket;

public:
    Pengiriman(string IdPengiriman, string StatusPengiriman, vector<Kurir> kurirList, int BeratPaket, string JenisPaket) {
        this->IdPengiriman = IdPengiriman;
        this->StatusPengiriman = StatusPengiriman;
        this->kurirList = kurirList;
        this->BeratPaket = BeratPaket;
        this->JenisPaket = JenisPaket;
    }



    string getIdPengiriman() {
        return IdPengiriman;
    }

    void setIdPengiriman(string IdPengiriman) {
        this->IdPengiriman = IdPengiriman;
    }

    string getStatusPengiriman() {
        return StatusPengiriman;
    }

    void setStatusPengiriman(string StatusPengiriman) {
        this->StatusPengiriman = StatusPengiriman;
    }

    vector<Kurir> getKurir() {
        return kurirList;
    }

    void setKurir(vector<Kurir> kurirList) {
        this->kurirList = kurirList;
    }

    int getBeratPaket() {
        return BeratPaket;
    }

    void setBeratPaket(int BeratPaket) {
        this->BeratPaket = BeratPaket;
    }

    string getJenisPaket() {
        return JenisPaket;
    }

    void setJenisPaket(string JenisPaket) {
        this->JenisPaket = JenisPaket;
    }
};

File: Python/Pengiriman.py
class Pengiriman:
    __IdPengiriman = ""
    __StatusPengiriman = ""
    __Kurir = []
    __BeratPaket = 0
    __JenisPaket = ""

    def __init__(self, IdPengiriman="", StatusPengiriman="", Kurir=None, BeratPaket=0, JenisPaket=""):
        self.__IdPengiriman = IdPengiriman
        self.__StatusPengiriman = StatusPengiriman
        self.__Kurir = Kurir if Kurir is not None else []
        self.__BeratPaket = BeratPaket
        self.__JenisPaket = JenisPaket
    
    def getIdPengiriman(self):
        return self.__IdPengiriman
    
    def setIdPengiriman(self, IdPengiriman):
        self.__IdPengiriman = IdPengiriman
    
    def getStatusPengiriman(self):
        return self.__StatusPengiriman
    
    def setStatusPengiriman(self, StatusPengiriman):
        self.__StatusPengiriman = StatusPengiriman

    def getKurir(self):
        return self.__Kurir
    
    def setKurir(self, Kurir):
        self.__Kurir = Kurir

    def getBeratPaket(self):
        return self.__BeratPaket
    
    def setBeratPaket(self, BeratPaket):
        self.__BeratPaket = BeratPaket

    def getJenisPaket(self):
        return self.__JenisPaket
    
    def setJenisPaket(self, JenisPaket):
        self.__JenisPaket = JenisPaket
